Lesson 2 - Functions/code/homework/code.js

```diff
@@ -23,23 +23,47 @@
     //////////////////////////////////////////////////////////////////////////////////////
 
     global.UAM.addAircraft = function (newAircraftCode) {
-        // function should return new aircraft object
+        var length = global.UAM.aircrafts.push({code: newAircraftCode, services: []});
+        
+        return global.UAM.aircrafts[length - 1];
     };
 
     global.UAM.removeAircraft = function (aircraftObj) {
-        // !!!
+        var pos = global.UAM.aircrafts.indexOf(aircraftObj);
+        
+        if(pos >= 0) {
+            global.UAM.aircrafts.splice(pos, 1);
+        }
     };
 
     global.UAM.addWorkToAircraft = function(aircraftObj, name, timeToExxecute) {
-        // !!!
+        aircraftObj.services.push({name: name, timeToExecute: timeToExxecute});
     };
         
-    global.UAM.reduceTimeToExecute = function(time) {
-        // !!!
+    global.UAM.reduceTimeToExecute = function(aircraftObj, time) {
+        if(aircraftObj.services !== null) {
```

Comment:
You should avoid using `null` and `undefined` in direct comparisons.
This is not an error but could be confusing.

Just `if(aircraftObj.services) {` - this will do.